mono/metadata/threads.c

```diff
@@ -5659,7 +5659,8 @@ mono_thread_is_foreign (MonoThread *thread)
 static void
 threads_native_thread_join_lock (gpointer tid, gpointer value)
 {
-	pthread_t thread = (pthread_t)tid;
+	/* This is a bit ugly, but handles platforms with small pthread_t. */
```

Comment:
No comment needed.

Reply:
Or, if you want a comment:
"pthread is a pointer or any size integer. Pointers can only be cast to/from pointer-sized integers. Integers can convert among any integers."

Reply:
There might be a macro for this.
Something like this:
https://github.com/modula3/cm3/blob/master/m3-libs/m3core/src/m3core.h#L407